actions/submit

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+set -ex
+
+if ! charms.reactive is_state 'spark.started'; then
+    action-fail 'Spark not yet ready'
+    exit
+fi
+
+# create dir to store results
+RESULT_DIR=/opt/sparkpi-jobs/$job_id
+RESULT_LOG=${RESULT_DIR}/job.log
+mkdir -p ${RESULT_DIR}
+chown -R ubuntu:ubuntu ${RESULT_DIR}
+
+py_files="$(action-get py-files)"
+packages="$(action-get packages)"
+extra_params="$(action-get extra-params)"
+class="$(action-get class)"
+job="$(action-get job)"
+job_args="$(action-get job-args)"
+cron="$(action-get cron)"
+
+submit_args='--master ${MASTER} --deploy-mode cluster'
+if [[ -n "$packages" ]]; then
+    submit_args="$submit_args --packages $packages"
+fi
+if [[ -n "$py_files" ]]; then
+    submit_args="$submit_args --packages $py_files"
+fi
+if [[ -n "$extra_params" ]]; then
+    submit_args="$submit_args $extra_params"
+fi
+if [[ -n "$class" ]]; then
+    submit_args="$submit_args --class $class"
+fi
+submit_args="$submit_args $job"
+
+job_code=". /etc/environment ; spark-submit ${submit_args} ${job_args}"
```

Comment:
Here is a crazy idea. If we had "spark-submit" as a parameter that defaults to "spark-submit" we could do more interesting things. For example, a user could upload to HDFS his script and then execute it using this action. Far more powerful, I think :)